pkg/kbagent/service/action_utils.go

```diff
@@ -150,11 +151,7 @@ func nonBlockingCallAction(ctx context.Context, action *kbaproto.Action, paramet
 func nonBlockingCallActionX(ctx context.Context, action *kbaproto.Action, parameters map[string]string, timeout *int32,
 	stdinReader io.Reader, stdoutWriter, stderrWriter io.Writer) (chan error, error) {
 	var cancel context.CancelFunc
-	if timeout == nil {
-		ctx, cancel = context.WithTimeout(ctx, defaultActionCallTimeout)
-	} else if *timeout > 0 {
-		ctx, cancel = context.WithTimeout(ctx, min(time.Duration(*timeout)*time.Second, maxActionCallTimeout))
-	}
+	ctx, cancel = actionCallTimeout(ctx, action.TimeoutSeconds, timeout)
```

Comment:
Why there are two timeout settings? Should the timeout defined in cmpd types be enough?